### PR TITLE
chat customization telemetry

### PR DESCRIPTION
**Pull request by Will-ShaoHua**
## verified from service side the following requests include the correct payload of customizationArn

```
// builderId
2024-07-24 11:21:36,113 [  54031]   FINE - software.aws.toolkits.jetbrains.services.cwc.controller.chat.telemetry.TelemetryHelper - Successfully sendTelemetryEvent for ChatInteractWithMessage with requestId=2fff6666-a03e-42cb-99c4-94f40220b872

2024-07-24 11:21:36,336 [  54254]   FINE - software.aws.toolkits.jetbrains.services.cwc.controller.chat.telemetry.TelemetryHelper - Successfully sendTelemetryEvent for ChatAddMessage with requestId=ef3f8751-1f3d-4db7-abc6-f84cf7694f83


2024-07-24 11:30:42,225 [ 600143]   FINE - software.aws.toolkits.jetbrains.services.codewhisperer.telemetry.CodeWhispererUserModificationTracker - Successfully sendTelemetryEvent for ChatModificationWithChat with requestId: 38ee3d14-557d-4c89-93b7-32cd981b68b5

// idc with customization NGDE-Custom-V13

2024-07-24 14:19:14,100 [ 291706]   FINE - software.aws.toolkits.jetbrains.services.cwc.controller.chat.telemetry.TelemetryHelper - Successfully sendTelemetryEvent for ChatAddMessage with requestId=d2ba6364-dcd0-4615-b352-06323c711235

2024-07-24 14:19:27,878 [ 305484]   FINE - software.aws.toolkits.jetbrains.services.cwc.controller.chat.telemetry.TelemetryHelper - Successfully sendTelemetryEvent for ChatInteractWithMessage with requestId=b02d9e58-920d-4e4b-9e3e-40ee334a297f

2024-07-24 14:25:26,602 [ 664208]   FINE - software.aws.toolkits.jetbrains.services.codewhisperer.telemetry.CodeWhispererUserModificationTracker - Successfully sendTelemetryEvent for ChatModificationWithChat with requestId: 198eb659-218d-4b29-a172-e39c8cad06a0


// idc without customization

2024-07-24 14:27:19,263 [ 776869]   FINE - software.aws.toolkits.jetbrains.services.cwc.controller.chat.telemetry.TelemetryHelper - Successfully sendTelemetryEvent for ChatAddMessage with requestId=c0a664d8-e2fc-4375-ae9a-acba08262c68

2024-07-24 14:27:42,577 [ 800183]   FINE - software.aws.toolkits.jetbrains.services.cwc.controller.chat.telemetry.TelemetryHelper - Successfully sendTelemetryEvent for ChatInteractWithMessage with requestId=ef5a3b66-3438-4fdd-888d-b36a206f7969



2024-07-24 14:33:30,606 [1148212]   FINE - software.aws.toolkits.jetbrains.services.codewhisperer.telemetry.CodeWhispererUserModificationTracker - Successfully sendTelemetryEvent for ChatModificationWithChat with requestId: c6ab16f2-6bea-4175-afe1-41bfc493060f
```

## Types of changes

- [ ] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)

## Checklist
- [ ] My code follows the code style of this project
- [ ] I have added tests to cover my changes
- [ ] A short description of the change has been added to the **[CHANGELOG](https://github.com/aws/aws-toolkit-jetbrains/blob/master/CONTRIBUTING.md#contributing-via-pull-requests)** if the change is customer-facing in the IDE.
- [ ] I have added metrics for my changes (if required)
 
## License
I confirm that my contribution is made under the terms of the Apache 2.0 license.

### CODE REVIEW

**Comment by leigaol** [comment]
Please update with the base branch and then fix the unit test.

**Review by zixlin7 — APPROVED** [review]

**Comment by zixlin7** [comment]
looks good pls fix the CI failures

File: plugins/amazonq/chat/jetbrains-community/src/software/aws/toolkits/jetbrains/services/cwc/controller/chat/telemetry/TelemetryHelper.kt [hunk 1]
```diff
@@ -3,15 +3,19 @@
 
 package software.aws.toolkits.jetbrains.services.cwc.controller.chat.telemetry
 
+import org.assertj.core.util.VisibleForTesting
```
Comment (rli) [inline plugins/amazonq/chat/jetbrains-community/src/software/aws/toolkits/jetbrains/services/cwc/controller/chat/telemetry/TelemetryHelper.kt:6]: ? wrong annotation

this library shouldnt even be available in this source context

File: plugins/amazonq/codewhisperer/jetbrains-community/tst/software/aws/toolkits/jetbrains/services/codewhisperer/CodeWhispererUserModificationTrackerTest.kt [hunk 2]
```diff
@@ -0,0 +1,181 @@
...
+import software.aws.toolkits.jetbrains.services.cwc.controller.chat.telemetry.InsertedCodeModificationEntry
+import software.aws.toolkits.jetbrains.services.telemetry.NoOpPublisher
+import software.aws.toolkits.jetbrains.services.telemetry.TelemetryService
+import java.time.Instant
+
+class CodeWhispererUserModificationTrackerTest {
+    private class NoOpToolkitTelemetryService(
+        publisher: TelemetryPublisher = NoOpPublisher(),
+        batcher: TelemetryBatcher
+    ) : TelemetryService(publisher, batcher)
```
Comment (rli) [inline plugins/amazonq/codewhisperer/jetbrains-community/tst/software/aws/toolkits/jetbrains/services/codewhisperer/CodeWhispererUserModificationTrackerTest.kt:53]: can you just make this a text fixture that can be reused so we dont keep copying it

File: plugins/amazonq/chat/jetbrains-community/tst/software/aws/toolkits/jetbrains/services/amazonq/TelemetryHelperTest.kt [hunk 3]
```diff
@@ -0,0 +1,647 @@
...
+import com.intellij.openapi.Disposable
+import com.intellij.openapi.application.ApplicationManager
+import com.intellij.openapi.project.Project
+import com.intellij.testFramework.ProjectExtension
+import com.intellij.testFramework.junit5.TestDisposable
+import com.intellij.testFramework.registerServiceInstance
+import com.intellij.testFramework.replaceService
+import kotlinx.coroutines.runBlocking
+import org.assertj.core.api.Assertions.assertThat
+import org.gradle.internal.impldep.com.amazonaws.ResponseMetadata.AWS_REQUEST_ID
```
Comment (rli) [inline plugins/amazonq/chat/jetbrains-community/tst/software/aws/toolkits/jetbrains/services/amazonq/TelemetryHelperTest.kt:15]: dont use internal gradle import

File: plugins/amazonq/chat/jetbrains-community/tst/software/aws/toolkits/jetbrains/services/amazonq/TelemetryHelperTest.kt [hunk 4]
```diff
@@ -0,0 +1,647 @@
...
+package software.aws.toolkits.jetbrains.services.amazonq
+
+import com.intellij.openapi.Disposable
+import com.intellij.openapi.application.ApplicationManager
+import com.intellij.openapi.project.Project
+import com.intellij.testFramework.ProjectExtension
+import com.intellij.testFramework.junit5.TestDisposable
+import com.intellij.testFramework.registerServiceInstance
+import com.intellij.testFramework.replaceService
+import kotlinx.coroutines.runBlocking
```
Comment (rli) [inline plugins/amazonq/chat/jetbrains-community/tst/software/aws/toolkits/jetbrains/services/amazonq/TelemetryHelperTest.kt:13]: avoid runBlocking in test

do
```
fun whatever() = runTest {
...
}

File: plugins/amazonq/chat/jetbrains-community/tst/software/aws/toolkits/jetbrains/services/amazonq/TelemetryHelperTest.kt [hunk 5]
```diff
@@ -0,0 +1,647 @@
...
+        // Toolkit telemetry
+        argumentCaptor<MetricEvent> {
+            verify(mockBatcher).enqueue(capture())
+            val event = allValues.first().data.find { it.name == "amazonq_addMessage" }
+
+            if (event == null) {
+                fail("amazonq_addMessage must not be null")
+            }
+
+            assertThat(event)
```
Comment (rli) [inline plugins/amazonq/chat/jetbrains-community/tst/software/aws/toolkits/jetbrains/services/amazonq/TelemetryHelperTest.kt:267]:
```
    assertThat(firstValue)
        .matches(...)
```

**Review by rli — APPROVED** [review]

**Pull request merged**